rollbar-java/src/main/java/com/rollbar/notifier/Rollbar.java

```diff
@@ -80,6 +89,43 @@ public void handleUncaughtErrors(Thread thread) {
   }
 
   /**
+   * Replace the configuration of this instance.
+   * This {@link ConfigBuilder} passed to configProvider is
+   * preconfigured with the values of the current configuration.
+   * This method potentially blocks to acquire a locks when
+   * safely work with the configuration.
+   *
+   * @param configProvider the provider of a new configuration
+   */
+  public void configure(ConfigProvider configProvider) {
+    ConfigBuilder builder;
+
+    this.configReadLock.lock();
+    try {
+      builder = ConfigBuilder.withConfig(this.config);
+    } finally {
+      this.configReadLock.unlock();
+    }
+
+    Config newConfig = configProvider.provide(builder);
+
+    this.configWriteLock.lock();
```

Comment:
I think we could use here the `configure(Config config)` method.
Also, I'd wrap all the operations after getting the lock in a `try` clause and the release of it in the `finally` clause. This could avoid some weird errors that could happen and its a kind of pattern that you always find, so makes easier to read the code, although it's more verbose.